src/ZODB/Connection.py

```diff
@@ -559,10 +497,6 @@ def _commit(self, transaction):
             elif oid in self._added:
                 assert obj._p_serial == z64
             elif obj._p_changed:
-                if oid in self._invalidated:
```

Comment:
Yup.  Plus it reduces commit work a little bit for the common case, so could be a net win. <shrug>